ocaml/xapi/message_forwarding.ml

```diff
@@ -1334,7 +1378,8 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
 							end
 						else
 							(* if we're nt suspended then just forward to host that has vm running on it: *)
-							forward_vm_op ~vm in
+							do_op_on ~host:host
```

Comment:
I suspect we don't need to change this now - you were probably hitting the exception before because the cancelled tasks weren't waiting for the events to be flushed.